util/dvsim/SGE.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env python
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+# ----------------------------------
+# SGE.py
+#       _JobData Class
+# ----------------------------------
+import logging
+import pwd
+import os
+import re
+import subprocess
+import time
+
+#
+from qsubopts import qsubOptions
+
+
+class _JobData:
+    """
+    Internal helper class to manage job data from qstat
+    """
+    def __init__(self, qstat_job_line):
+        # The format of the line goes like
+        # job-ID prior name  user  state submit/start at queue slots ja-task-ID
+
+        tokens = qstat_job_line.split()
+
+        assert len(tokens) >= 8, 'Not a valid qstat line: ' + qstat_job_line
+        # Job line must have at least 8 tokens
+        self.id = int(tokens[0])
+        self.priority = float(tokens[1])
+        self.name = tokens[2]
+        self.user = tokens[3]
+        self.state = tokens[4]
+        self.time = ' '.join(tokens[5:7])
+
+        if '@' in qstat_job_line:
+            # Has queue assigned, e.g. core2.q@q2.caspian.mit.edu
+            self.queue = tokens[7]
+            self.slots = tokens[8]
+            if len(tokens) == 9:
+                self.ja_task_ID = None
+            elif len(tokens) == 10:
+                self.ja_task_ID = tokens[-1]
```

Comment:
It would be clearer to say tokens[9] since you know there are 10 tokens.

Reply:
Matute thank you for the review and comments. I was on vacation so only now got to it.
The tokens represents the queue status (qstat_job_line) and we can't know the length of the status that will be returned.